lib/HLSL/HLMatrixLowerPass.cpp

```diff
@@ -683,6 +684,19 @@ Value *HLMatrixLowerPass::lowerCall(CallInst *Call) {
     ? lowerNonHLCall(Call) : lowerHLOperation(Call, OpcodeGroup);
 }
 
+// Special function to lower precise call applied to a matrix
+// The matrix should be lowered and the call regenerated with vector arg
+void HLMatrixLowerPass::lowerPreciseCall(CallInst *Call, IRBuilder<> Builder) {
+  DXASSERT(Call->getNumArgOperands() == 1, "Only one arg expected for precise matrix call");
+  Value *Arg = Call->getArgOperand(0);
+  Value *LoweredArg = getLoweredByValOperand(Arg, Builder);
+  for (Value *A : Call->arg_operands()) {
+    DXASSERT(A == Arg, "oops");
```

Comment:
How is this necessary?  There's already:
```
DXASSERT(Call->getNumArgOperands() == 1, ...
```
and
```
Value *Arg = Call->getArgOperand(0);
```
So how could A be different unless there's a fundamental problem with operand iteration or something?

Reply:
Debug code I meant to remove. Hang on a sec.